From 84aae469dba83c7d8b6bec938007243757ea8aa1 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
Date: Mon, 19 May 2025 11:17:11 +0800
Subject: [PATCH 04/43] riscv: k1: dt-bindings: support dma binding for
 spacemit k1 soc

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/25
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/f8496e87c2b30603519d8303a32536eec14f3eda

--------------------------------

Add dma bindings for Spacemit k1 SoC dma controller, it
defines the drq number for various devices

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
---
 include/dt-bindings/dma/spacemit-k1-dma.h | 54 +++++++++++++++++++++++
 1 file changed, 54 insertions(+)
 create mode 100644 include/dt-bindings/dma/spacemit-k1-dma.h

diff --git a/include/dt-bindings/dma/spacemit-k1-dma.h b/include/dt-bindings/dma/spacemit-k1-dma.h
new file mode 100644
index 0000000000000..43b6904b912a3
--- /dev/null
+++ b/include/dt-bindings/dma/spacemit-k1-dma.h
@@ -0,0 +1,54 @@
+/* SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause */
+/*
+ * Request type numbers for the Spacemit k1 DMA.
+ *
+ * Copyright (C) 2023 Spacemit
+ */
+
+#ifndef __DT_BINDINGS_DMA_SMACEMIT_K1_H__
+#define __DT_BINDINGS_DMA_SMACEMIT_K1_H__
+
+#define DMA_UART0_TX	3
+#define DMA_UART0_RX	4
+#define DMA_UART2_TX	5
+#define DMA_UART2_RX	6
+#define DMA_UART3_TX	7
+#define DMA_UART3_RX	8
+#define DMA_UART4_TX	9
+#define DMA_UART4_RX	10
+#define DMA_I2C0_TX	11
+#define DMA_I2C0_RX	12
+#define DMA_I2C1_TX	13
+#define DMA_I2C1_RX	14
+#define DMA_I2C2_TX	15
+#define DMA_I2C2_RX	16
+#define DMA_I2C4_TX	17
+#define DMA_I2C4_RX	18
+#define DMA_SSP3_TX	19
+#define DMA_SSP3_RX	20
+#define DMA_SSPA0_TX	21
+#define DMA_SSPA0_RX	22
+#define DMA_SSPA1_TX	23
+#define DMA_SSPA1_RX	24
+#define DMA_UART5_TX	25
+#define DMA_UART5_RX	26
+#define DMA_UART6_TX	27
+#define DMA_UART6_RX	28
+#define DMA_UART7_TX	29
+#define DMA_UART7_RX	30
+#define DMA_UART8_TX	31
+#define DMA_UART8_RX	32
+#define DMA_UART9_TX	33
+#define DMA_UART9_RX	34
+#define DMA_I2C5_TX	35
+#define DMA_I2C5_RX	36
+#define DMA_I2C6_TX	37
+#define DMA_I2C6_RX	38
+#define DMA_I2C7_TX	39
+#define DMA_I2C7_RX	40
+#define DMA_I2C8_TX	41
+#define DMA_I2C8_RX	42
+#define DMA_QSPI_RX	44
+#define DMA_QSPI_TX	45
+
+#endif /* __DT_BINDINGS_DMA_SMACEMIT_K1_H__ */
-- 
2.51.0

